src/backend/main.ts

```diff
@@ -160,6 +165,76 @@ export async function bootstrapWorkspace(opts: {
+  console.log('debug: nodeID', nodeId)
+  const snapshotName = appPrefix.append('32=snapshot').toString()
+  console.log('debug: theoretical snapshot prefix: ', snapshotName)
+  const localYJSUpdate = await persistStore.get('localSnapshot')
+  console.log('persistent storage local yjs update: ', localYJSUpdate)
+  // localYJSUpdate = undefined
+
+  // //State Vector Testing
+  // let test = new StateVector()
+  // test.set(new Name('/user1'), 1)
+  // test.set(new Name('/user2'), 1)
+  // let svsComponent = new Component(Version.type, Encoder.encode(test))
+  // console.log('Debug: SVS test: ', svsComponent.toString())
+  // console.log('Debug: parsing SVS', Decoder.decode(svsComponent.value, StateVector))
+
+  if (!localYJSUpdate) {
+    console.log('Empty yjs local storage detected, fetching snapshot')
+    const interest = new Interest(snapshotName, Interest.CanBePrefix, Interest.MustBeFresh)
+    try {
+      const data = await consume(interest)
+      console.log(`Received data with name [${data.name}]`)
+      const targetName = data.name.getPrefix(-1)
+      // /grpPrefix/32=snapshot/54=<vector>/
+      console.log('segmented object fetcher targeting name: ', targetName.toString())
+      // code from SyncAgent
+      const buffers = []
+      try {
+        console.log('SegmentedObject fetching a snapshot at part 4')
+        const result = fetch(targetName, {
+          verifier: certStorage.verifier, // we have access to the verifier
+          modifyInterest: { mustBeFresh: true },
+          lifetimeAfterRto: 2000,
+          retxLimit: 150, // See Deliveries. 60*1000/(2*200)=150. Default minRto = 150.
+        })
+        for await (const segment of result) {
+          // Cache packets temporary disabled.
+          // this.persistStorage.set(segment.name.toString(), Encoder.encode(segment));
+          // Reassemble
+          buffers.push(segment.content)
+        }
+      } catch (e: any) {
+        console.error(`Unable to fetch ${targetName}: `, e)
+      }
+      const snapshotData = concatBuffers(buffers) //uint8array
+      persistStore.set('localSnapshot', snapshotData)
+      // TODO: SVS parsing check TLV type.
+      persistStore.set('localState', targetName.at(-1).value)
+      let aloSyncKey = '/8=local' + nodeId.toString() + '/32=sync/32=alo/8=syncVector'
+      console.log('targeting alo sync key: ', aloSyncKey)
+      // 8=local/8=ndn-workspace/8=test/8=node-1/32=sync/32=alo/8=syncVector
+      persistStore.set(aloSyncKey, targetName.at(-1).value)
+      // debug: check state vector count
+      let decodedSV = Decoder.decode(targetName.at(-1).value, StateVector)
+      let count = 0
+      for (const [id, seq] of decodedSV) {
+        count += seq
+      }
+      console.log('Written the following total state vector count into persistent storage:', count)
+      console.log('debug: state vector: ', targetName.at(-1).value)
+      await new Promise((r) => setTimeout(r, 1000))
```

Comment:
1s would be a significant overhead on loading. Let's figure out what needs to be waited after you finish implementation.

Reply:
The listener is called "after" the blob segment fetcher "finish" fetching the snapshot from sync. 
Sometimes when the code gets to persistentObject.get() it would return empty, which suggests the write of the fetched snapshot is incomplete but then the listener was still called.
A better way to handle this is to check if the snapshot exists in persiststorage and somehow wait til it does.
As noted in the code, I previously tried calling getBlob() but that actually results in the segmented fetcher being called up again.
1s is playing it safe. Actually if the segmented fetcher has problems upstream, even this amount of wait would actually fail, hence I have an additional catch to give up updating the blind fetch response, if that happens. 
I might even have to change how that blind fetch response system is updated, if it comes to that.

Reply:
`persistStore.set` should be awaitable. I did not do this in SyncAgent because I don't want to block further fetch.

Reply:
Here is something strange: 
I tried awaiting the persistStore.set().. It loads snapshot correctly but editor stopped synchronizing after that (but then sync interest and snapshots still go through).
And even if i roll back it still have a problem, and I have to restart my local NFD to get it working again.
I will keep the 1000ms there, until we root cause this.

Reply:
![image](https://github.com/user-attachments/assets/c3b692dd-01f6-4598-bcf4-b198b9c45591)
![image](https://github.com/user-attachments/assets/7a586be3-a2a9-4ea5-8312-a6cb90909aad)
![image](https://github.com/user-attachments/assets/793b9223-06c1-4e17-bf39-9d0af51e9e1c)
![image](https://github.com/user-attachments/assets/2d2c3902-dc5f-4f5d-9aaa-99a79e8f685b)